core/templates/services/literally-canvas-helper.service.ts

```diff
@@ -0,0 +1,190 @@
+// Copyright 2019 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview LiterallyCanvas editor helper service.
+ */
+
+
+angular.module('oppia').factory('LiterallyCanvasHelperService', [
+  function() {
+    const getPoints = (x, y, angle, width, length) => [
+      {
+        x: x + (Math.cos(angle + Math.PI / 2) * width) / 2,
+        y: y + (Math.sin(angle + Math.PI / 2) * width) / 2,
+      },
+      {
+        x: x + Math.cos(angle) * length,
+        y: y + Math.sin(angle) * length,
+      },
+      {
+        x: x + (Math.cos(angle - Math.PI / 2) * width) / 2,
+        y: y + (Math.sin(angle - Math.PI / 2) * width) / 2,
+      },
+    ];
+    const arrow = {
+      svg(x, y, angle, width, color, position, length = null) {
+        if (length === null) {
+          length = 0;
+        }
+        length = length || width;
+        const points = getPoints(x, y, angle, width, length);
+
+        return ('<polygon id="' + position + '" fill="' + color + '" ' +
+        'stroke="none" points="' + (points.map(function(p) {
+          return p.x + ',' + p.y;
+        }).join(' ')) + '" />');
+      }
+    };
+    return {
+      rectangleSVGRenderer: function(shape) {
+        // This function converts a rectangle shape object to the rect tag.
+        var height, width, x, x1, x2, y, y1, y2, id;
+        x1 = shape.x;
+        y1 = shape.y;
+        x2 = shape.x + shape.width;
+        y2 = shape.y + shape.height;
+        x = Math.min(x1, x2);
+        y = Math.min(y1, y2);
+        width = Math.max(x1, x2) - x;
+        height = Math.max(y1, y2) - y;
+        if (shape.strokeWidth % 2 !== 0) {
+          x += 0.5;
+          y += 0.5;
+        }
+        id = 'rectangle-' + shape.id;
+        return ('<rect id="' + id + '" x="' + x + '" y="' + y + '" width="' +
```

Comment:
This won't work, sorry -- we don't allow construction of html in JS, and definitely not with "+" concatenation, since it is vulnerable to programming errors leading to security holes. E.g. maybe somehow shapeId has `</rect></svg><script src="evil.com">` in it. /cc @marianazangrossi 

I think you'll need to create a rect element and use e.g. jQuery to add attributes to it, or something like that. Make sure to add unit tests for sanitization (try using "bad" shapeIDs and ensure they are correctly escaped when the tag is produced.

Ditto elsewhere.

Reply:
Just to add to this please either use something that AngularJS or Angular 8 provides for construction HTML or pure JavaScript techniques, we shouldn't add more code that is dependent on jQuery directly.

Reply:
I have used the document to create svg tags and also added validations for it. Done.